google/cloud/storage/internal/curl_handle.cc

```diff
@@ -226,26 +226,36 @@ Status CurlHandle::AsStatus(CURLcode e, char const* where) {
     case CURLE_NOT_BUILT_IN:
       code = StatusCode::kUnknown;
       break;
+
     case CURLE_COULDNT_RESOLVE_PROXY:
     case CURLE_COULDNT_RESOLVE_HOST:
     case CURLE_COULDNT_CONNECT:
-    case CURLE_WEIRD_SERVER_REPLY:
       code = StatusCode::kUnavailable;
       break;
+
+#if LIBCURL_VERSION_NUM >= 0x075100
+    case CURLE_WEIRD_SERVER_REPLY:
+      code = StatusCode::kUnknown;
+      break;
+#endif  // CURL >= 7.51.0
```

Comment:
It seems that 7.51.0 would be represented as 0x073300.

But then again, why shouldn't this just be handled by ...
```
      // There are about 100 error codes, some are explicitly marked as
      // obsolete, some are not available in all libcurl versions. Use this
      // `default:` case to treat all such errors as `kUnknown`.
```

Reply:
Good point. Let me give it another shot.

I find it hard to discover which error codes I thought about before and which ones I did not, so I listed all of them (they add new ones from time to time, more than 10 since I last looked), so I wanted to list all of them explicitly. But cannot really list "all" because some do not show up with older version.

Let me know what you think of the new approach.